### PR TITLE
Xpack Role and RoleMapping for ES provider

### PR DESCRIPTION
**Pull request by Enteris**
Hey @phillbaker 
I managed to embark on the way the test were run in your repo and fix the travis.yml

A few things to note : 

- I added a specific check for every ES type we launch in a version. It's ugly, feel free to improve if you wish.
- I was using GoRelease in my branch to push releases, I saw you guys used the release provider, feel free to remove one of them.
- I added stages support in Travis.
- I merged with your updated support for ES7, and since ES5 is hardly testable, it's not supported.

Just ping me if you have any questions about the merge.

### CODE REVIEW

**Comment by Enteris** [comment]
Oh, also, maybe we should push the Release key in a secure env var in the Travis-ci instead of hardcoding it in the travis.yml, but I'll let you handle this as I probably don't have access.

File: provider.go [hunk 1]
```diff
@@ -32,6 +32,18 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("ELASTICSEARCH_URL", nil),
 				Description: "Elasticsearch URL",
 			},
+			"sniff": &schema.Schema{
```
Comment (phillbaker) [inline provider.go]: Can you pull this out into a separate PR? Happy to get this merged quickly.

File: .travis.yml [hunk 2]
```diff
@@ -31,12 +31,15 @@ install:
   - env GO111MODULE=on go mod vendor
 script:
   - export GO111MODULE=on
-  - wget -q --waitretry=1 --retry-connrefused -T 30 -O - $ELASTICSEARCH_URL
+  - wget -q --waitretry=5 --tries=50 --retry-connrefused -T 600 -O - http://127.0.0.1:9200
```
Comment (phillbaker) [inline .travis.yml]: Aha! It looks like the `--tries=50` is the key to getting this to work on travis! Nice!

File: provider.go (hunk 1 above)
Comment (Enteris) [reply provider.go]: The Sniff and Healthcheck support? I will. It's useful on a setup where the nodes aren't routable :+1:

File: .travis.yml (hunk 2 above)
Comment (Enteris) [reply .travis.yml]: Yeah, our issue was that wget was reaching 20 tries in 20sec, so it never waited 30 sec. I bumped a few parameters, 10min should never happen and might be overkill but we could always lower it a bit.
